packages/next/build/webpack-config.ts

```diff
@@ -495,6 +504,23 @@ export default async function getBaseWebpackConfig(
               'next/constants',
             ]
 
+            // next/head import should not be allowed inside _document page
+            // So we check for _document content and throw an error in case next/head is imported
+            if (request === 'next/head' && customDocumentFile) {
```

Comment:
This expression will always be true even in pages that are not _document. The right approach would be to traverse up the request chain through `issuer`, not sure if that's exposed to the externals function though 🤔